syncer/checkpoint.go

```diff
@@ -226,28 +229,30 @@ func NewRemoteCheckPoint(tctx *tcontext.Context, cfg *config.SubTaskConfig, id s
 }
 
 // Init implements CheckPoint.Init
-func (cp *RemoteCheckPoint) Init(conn *Conn) error {
+func (cp *RemoteCheckPoint) Init(conn *DBConn) error {
```

Comment:
this `conn *DBConn` only used for testing? Is there any better method to do that?

Reply:
it's only used for testing, also `ShardingGroupKeeper.Init()`.
A better idea may be 
1.  change `cp.db` to `cp.DB`
2. add setDB function
which one is better?

Reply:
I prefer `add setDB function`, or can we assign `cp.db` directly in the tests after converted with `cp.(*RemoteCheckPoint)`?

Reply:
ok, let my try assign cp.db first

Reply:
I removed conn argument from checkPoint  [`Init()`](https://github.com/pingcap/dm/pull/266/files#diff-ef22d46ca0482ca5b7cf503016990902R232), and also shardGroupKeeper `Init()`, in order to remove it, I moved `sgk.Init()` out of [`initShardGroup`](https://github.com/pingcap/dm/pull/266/files#diff-ea4abf20044394891136e70431919a4fL412) function, please check it